src/payment/types.rs

```diff
@@ -474,3 +477,475 @@ impl StrictDecode for ShortChannelId {
+                d.read_exact(&mut port[..])?;
+                let port = u16::from_be_bytes(port);
+
+                Ok(AnnouncedNodeAddr::IPv6 {
+                    addr: addr,
+                    port: port,
+                })
+            }
+
+            3u8 => {
+                let mut addr = [0u8; 10];
+                let mut port = [0u8; 2];
+                d.read_exact(&mut addr[..])?;
+                d.read_exact(&mut port[..])?;
+                let port = u16::from_be_bytes(port);
+
+                Ok(AnnouncedNodeAddr::OnionV2 {
+                    addr: addr,
+                    port: port,
+                })
+            }
+
+            4u8 => {
+                let mut ed2559_pubkey = [0u8; 32];
+                let mut checksum = [0u8; 2];
+                let mut version = [0u8; 1];
+                let mut port = [0u8; 2];
+                d.read_exact(&mut ed2559_pubkey[..])?;
+                d.read_exact(&mut checksum[..])?;
+                d.read_exact(&mut version[..])?;
+                d.read_exact(&mut port[..])?;
+                let checksum = u16::from_be_bytes(checksum);
+                let version = u8::from_be_bytes(version);
+                let port = u16::from_be_bytes(port);
+
+                Ok(AnnouncedNodeAddr::OnionV3 {
+                    ed25519_pubkey: ed2559_pubkey,
+                    checksum: Some(checksum),
+                    version: Some(version),
+                    port: port,
+                })
+            }
+
+            _ => Err(strict_encoding::Error::UnsupportedDataStructure(
+                "Wrong Network Address Format",
+            )),
+        }
+    }
+}
+
+impl lightning_encoding::Strategy for AnnouncedNodeAddr {
+    type Strategy = lightning_encoding::strategies::AsStrict;
+}
+
+#[derive(Wrapper, Clone, Debug, Display, From, PartialEq, Eq)]
+#[display(Debug)]
+pub struct AddressList(Vec<AnnouncedNodeAddr>);
+
+impl StrictEncode for AddressList {
+    fn strict_encode<E: io::Write>(
+        &self,
+        mut e: E,
+    ) -> Result<usize, strict_encoding::Error> {
+        let mut written = 0;
+        let len = self.0.len() as u16;
+        written += e.write(&len.to_be_bytes()[..])?;
+        for addr in &self.0 {
+            written += addr.strict_encode(&mut e)?;
+        }
+        Ok(written)
+    }
+}
+
+impl StrictDecode for AddressList {
+    fn strict_decode<D: io::Read>(
+        mut d: D,
+    ) -> Result<Self, strict_encoding::Error> {
+        let mut len_bytes = [0u8; 2];
+        d.read_exact(&mut len_bytes)?;
+        let len = u16::from_be_bytes(len_bytes) as usize;
+        let mut data = Vec::<AnnouncedNodeAddr>::with_capacity(len);
+        for _ in 0..len {
+            data.push(AnnouncedNodeAddr::strict_decode(&mut d)?);
+        }
+        Ok(AddressList(data))
+    }
+}
+
+impl lightning_encoding::Strategy for AddressList {
+    type Strategy = lightning_encoding::strategies::AsStrict;
```

Comment:
This will not work since the size of the vec will be encoded little-endian and  not big-endian

Reply:
Please check my comment below.